src/librustdoc/html/static/main.js

```diff
@@ -2950,8 +2938,46 @@ function defocusSearchBar() {
         buildHelperPopup = function() {};
     }
 
+    function loadScript(url) {
+        var script = document.createElement('script');
+        script.src = url;
+        document.head.append(script);
+    }
+
+    function setupSearchLoader() {
+        var searchLoaded = false;
+        function loadSearch() {
+            if (!searchLoaded) {
+                searchLoaded = true;
+                loadScript(window.searchJS);
+            }
+        }
+
+        // Assumes `crates{version}.js` has already been loaded, setting window.ALL_CRATES.
+        addSearchOptions(window.ALL_CRATES);
+        addSidebarCrates(window.ALL_CRATES);
+
+        search_input.addEventListener("focus", function(){
+            search_input.origPlaceholder = search_input.placeholder;
+            search_input.placeholder = "Type your search here.";
```

Comment:
I'm not sure to understand why you're updating the placeholder when the input gets the focus. Any reason?

Reply:
This is part of my attempts to make the search feel more responsive. Even though it might take some time for the search index to load, and for your query to run, it's nice for the page to immediately respond to your input in some way. This is the "response" idea from [Response, Animation, Idle, Load (RAIL)](https://web.dev/rail/).

Reply:
Tested it and I actually like how it reacts so let's keep it! :)